src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -94,6 +100,10 @@ public ExamResource(UserService userService, ExamService examService, ExamAccess
             return createExam(courseId, updatedExam);
         }
 
+        if (updatedExam.getCourse() == null) {
```

Comment:
Should exam here actually have a course? I mean the courseid is already defined through the path variable, isn't it?
If the issue would be that the provided entity is wrong you should consider HTTP 422 Unprocessable Entity
If considering, adapt tests as well.

Reply:
From comment above:

> Yeah it should have a course, otherwise we have to set the course manually before we do `examService.save(exam)`. Still, it's good that you commented here. We should check if the course of the delivered exam matches the `courseId` in the route as we check if the user is at least instructor in the course with `courseId`. I will adapt that.

Here it makes even more sense to check for the course as the client receives the exam object with the course and the client will send the updated exam back.